From 1e848a3e6f98e65064ae24f364dc877ae6bfa2b9 Mon Sep 17 00:00:00 2001
From: CMY <cmy@rock-chips.com>
Date: Wed, 27 Aug 2014 11:25:46 +0800
Subject: [PATCH] rk: memory: set vmalloc size to 496MB by default, so:   1. if
 memory is 256MB, then: 256MB(lowmem)+744MB(vmalloc)+0MB(highmem)   2. if
 memory is 512MB, then: 512MB(lowmem)+488MB(vmalloc)+0MB(highmem)   3. if
 memory is large then 512MB, then: 512MB(lowmem)+488MB(vmalloc)+other(highmem)

---
 arch/arm/boot/dts/rk312x-sdk.dtsi      | 2 +-
 arch/arm/boot/dts/rk3288-p977.dts      | 4 ----
 arch/arm/boot/dts/rk3288-p977_8846.dts | 4 ----
 arch/arm/boot/dts/rk3288-tb.dts        | 3 ---
 arch/arm/boot/dts/rk3288.dtsi          | 4 ++++
 5 files changed, 5 insertions(+), 12 deletions(-)

diff --git a/arch/arm/boot/dts/rk312x-sdk.dtsi b/arch/arm/boot/dts/rk312x-sdk.dtsi
index 35c23765dd79..e98de5af4cbf 100755
--- a/arch/arm/boot/dts/rk312x-sdk.dtsi
+++ b/arch/arm/boot/dts/rk312x-sdk.dtsi
@@ -9,7 +9,7 @@
 	};
 
 	chosen {
-		bootargs = "vmalloc=512M clk_ignore_unused";
+		bootargs = "vmalloc=496M clk_ignore_unused";
 	};
 
 	pwm_regulator1:pwm-regulator1 {
diff --git a/arch/arm/boot/dts/rk3288-p977.dts b/arch/arm/boot/dts/rk3288-p977.dts
index 6d74a360cd04..9452c1aff290 100755
--- a/arch/arm/boot/dts/rk3288-p977.dts
+++ b/arch/arm/boot/dts/rk3288-p977.dts
@@ -152,10 +152,6 @@
 		rockchip,remote_wakeup;
 		rockchip,usb_irq_wakeup;
 	};
-
-        chosen {
-                bootargs = "vmalloc=512M";
-        };
 };
 &pinctrl {
 	gpio0_gpio {
diff --git a/arch/arm/boot/dts/rk3288-p977_8846.dts b/arch/arm/boot/dts/rk3288-p977_8846.dts
index 3e258850da65..512e0946a624 100755
--- a/arch/arm/boot/dts/rk3288-p977_8846.dts
+++ b/arch/arm/boot/dts/rk3288-p977_8846.dts
@@ -155,10 +155,6 @@
 		rockchip,remote_wakeup;
 		rockchip,usb_irq_wakeup;
 	};
-
-        chosen {
-                bootargs = "vmalloc=512M";
-        };
 };
 &pinctrl {
 	gpio0_gpio {
diff --git a/arch/arm/boot/dts/rk3288-tb.dts b/arch/arm/boot/dts/rk3288-tb.dts
index 2f99f0b032db..097ba0ea1d08 100755
--- a/arch/arm/boot/dts/rk3288-tb.dts
+++ b/arch/arm/boot/dts/rk3288-tb.dts
@@ -171,9 +171,6 @@
 		rockchip,remote_wakeup;
 		rockchip,usb_irq_wakeup;
 	};
-	chosen {
-                bootargs = "vmalloc=512M";
-        };
 };
 
 &gmac {
diff --git a/arch/arm/boot/dts/rk3288.dtsi b/arch/arm/boot/dts/rk3288.dtsi
index 392026378dcc..b837e4765f5a 100755
--- a/arch/arm/boot/dts/rk3288.dtsi
+++ b/arch/arm/boot/dts/rk3288.dtsi
@@ -1364,4 +1364,8 @@
 		pinctrl-1 = <&sdcard_vcc_18>;
 		pinctrl-2 = <&sdcard_vcc_33>;
 	};
+
+        chosen {
+                bootargs = "vmalloc=496M";
+        };
 };
-- 
2.35.3

